### PR TITLE
Add two stacks method to sliding window section

### PR DESCRIPTION
**Pull request by ChuangSheep**
I think cpalgo explained the method quite well. But if u wish to have a short explanation of the method, I could write some. 

---

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/4_Gold/Sliding_Window.mdx [hunk 1]
```diff
@@ -295,7 +295,189 @@ def maxSlidingWindow(self, nums: List[int], k: int) -> list[int]:
...
+		}
+
+		return maxVals;
+	}
+
+	private static class MaxQueue {
+		/*
+		 * For each Pair e, e.a is the value of the element and
+		 * e.b is the maximum among all elements in that stack under element e.
+		 */
```
Comment (SansPapyrus683) [inline content/4_Gold/Sliding_Window.mdx:415]:
```suggestion
		/**
		 * For each Pair e, e.a is the value of the element and
		 * e.b is the maximum among all elements in that stack under element e.
		 */
```
doc comments start w/ 2 asterisks (same for c++ and stuff)

File: content/4_Gold/Sliding_Window.mdx [hunk 2]
```diff
@@ -295,7 +295,189 @@ def maxSlidingWindow(self, nums: List[int], k: int) -> list[int]:
...
+
+		/*
+		 * Remove the first element from our MaxQueue by poping the top element from
+		 * s2. Since elements in s1 are stored in the order of how we added them, i.e.
+		 * the last added element is at the top of s1, we just have to pop all of them
+		 * out and push them into the stack s2. After that, these elements in s2 will
+		 * be in reversed order, i.e. the first added element will be at the top of s2
+		 * and we can pop them out as from normal stacks. We only move elements from
+		 * s1 to s2 when s2 is empty.
+		 */
```
Comment (SansPapyrus683) [inline content/4_Gold/Sliding_Window.mdx:453]: at this point just put it above in an explanation, comments in guide code shouldn't be this long

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Sliding_Window.mdx [hunk 3]
```diff
@@ -295,7 +295,183 @@ def maxSlidingWindow(self, nums: List[int], k: int) -> list[int]:
 
 Method 3 from cp-algo. Not as common but nice to know!
 
-<IncompleteSection />
+We use two stacks $s1$ and $s2$ to simulate our maximum queue. Every time we add an element to it, we push the element itself and the maximum in stack $s1$ after adding this element to $s1$. To pop out the front element from our queue, we just pop the top element from $s2$. Since elements in $s1$ are stored in the order of how we added them, i.e. the last added element is at the top of $s1$, we just have to pop all of them out and push them into the stack $s2$ when $s2$ is empty. After that, these elements in $s2$ will be in reversed order, i.e. the first added element will be at the top of $s2$, and we can pop them out as from normal stacks to simulate the dequeue operation of our queue. To find the maximum among all elements in our queue, we just have to return the maximum of both stacks, which is stored in the top element when we added it.
```
Comment (SansPapyrus683) [inline content/4_Gold/Sliding_Window.mdx:298]: `$s_1$` instead of `$s1$`

File: content/4_Gold/Sliding_Window.mdx [hunk 4]
```diff
@@ -295,7 +295,183 @@ def maxSlidingWindow(self, nums: List[int], k: int) -> list[int]:
...
 
-<IncompleteSection />
+We use two stacks $s1$ and $s2$ to simulate our maximum queue. Every time we add an element to it, we push the element itself and the maximum in stack $s1$ after adding this element to $s1$. To pop out the front element from our queue, we just pop the top element from $s2$. Since elements in $s1$ are stored in the order of how we added them, i.e. the last added element is at the top of $s1$, we just have to pop all of them out and push them into the stack $s2$ when $s2$ is empty. After that, these elements in $s2$ will be in reversed order, i.e. the first added element will be at the top of $s2$, and we can pop them out as from normal stacks to simulate the dequeue operation of our queue. To find the maximum among all elements in our queue, we just have to return the maximum of both stacks, which is stored in the top element when we added it. 
+
+Then, we can solve the problem intuitively by removing the first element and adding a new element to the queue to simulate our sliding window. As each operation of our queue takes $\mathcal{O}(1)$ time, and we add each of $N$ elements once, we get a time complexity of $\mathcal{O}(N)$.
```
Comment (SansPapyrus683) [inline content/4_Gold/Sliding_Window.mdx:300]: remove "intuitively"

File: content/4_Gold/Sliding_Window.mdx [hunk 5]
```diff
@@ -295,7 +295,183 @@ def maxSlidingWindow(self, nums: List[int], k: int) -> list[int]:
...
+					int mx = s2.empty() ? s1.peek().a : s2.peek().b;
+					s2.push(new Pair(s1.peek().a, Math.max(mx, s1.peek().a)));
+					s1.pop();
+				}
+			}
+			s2.pop();
+		}
+	}
+
+	// data structure to store two integers
```
Comment (SansPapyrus683) [inline content/4_Gold/Sliding_Window.mdx:461]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — COMMENTED** [review]
@envyaims

**Review by envyaims — APPROVED** [review]
nice, pretty applaudable stuff :D

**Pull request merged**